lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -77,10 +79,14 @@ export async function generateLockFile(
     logger.trace({ cmd, args }, 'pnpm command');
     commands.push(`${cmd} ${args}`);
 
+    const pnpmVersion = execSync('pnpm --version').toString().trim();
```

Comment:
It might not be installed by here so this won't work reliably

Reply:
Considering your point about `pnpm` potentially not being installed at this stage, I propose using the existing `getPackageManagerVersion` method to retrieve the `pnpm` version from the `package.json`. This method should provide us with a more robust mechanism.

In cases where the `engines` field is missing in the `package.json`, or when we can't get a valid version for other reasons, I plan to not pass the `--ignore-scripts` option when executing `pnpm dedupe`.

Let me know your thoughts on this approach.